### PR TITLE
Reset password in Windows after it expires

### PR DESCRIPTION
**Pull request by pablo-herranz**
From some point to now, our password is periodically expired, even though we set the flag for it to never happen. Here's a workaround in case it expires again

- Related ticket: https://progress.opensuse.org/issues/153105
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/-/blob/74ebbbfeb4293f582cacc3ba6728f2838e76fe3e/win-password-expired-20240325.png
- Verification run:
  - https://openqa.suse.de/tests/13860100
  - https://openqa.suse.de/tests/13860101
  - https://openqa.suse.de/tests/13860114

### CODE REVIEW

File: lib/windowsbasetest.pm [hunk 1]
```diff
@@ -237,4 +246,15 @@ sub power_configuration {
     );
 }
 
+sub password_expired {
+    # There's need to type the old password and the new one twice and then press
+    # enter an additional time for acknowledging.
+    send_key 'ret';    # Ok message for starting the process
+    my @i = (0 .. 2);
+    for (@i) {
```
Comment (ricardobranco777) [inline lib/windowsbasetest.pm:254]: Nitpicking here but apply only if we don't need to restart VR.

```suggestion
    foreach my $i (0 .. 2) {
```

**Review by ricardobranco777 — APPROVED** [review]

File: lib/windowsbasetest.pm (hunk 1 above)
Comment (okurz) [reply lib/windowsbasetest.pm:254]: Actually you don't use the iterator so you can just use

```suggestion
    for (0 .. 2) {
```

File: lib/windowsbasetest.pm [hunk 2]
```diff
@@ -131,7 +131,10 @@ sub wait_boot_windows {
     assert_screen 'windows-screensaver', 900;
     send_key_until_needlematch 'windows-login', 'esc';
     type_password;
-    send_key 'ret';    # press shutdown button
+    send_key 'ret';
+    # Once a month Windows passwords expire, although we try to set it to never
+    # expire.
+    $self->password_expired if (check_screen('win-passwd-expired'));
```
Comment (okurz) [inline lib/windowsbasetest.pm:137]:
```suggestion
    $self->password_expired if check_screen 'win-passwd-expired';
```

File: lib/windowsbasetest.pm (hunk 1 above)
Comment (pablo-herranz) [reply lib/windowsbasetest.pm]: Thanks for your suggestions :)

**Review by okurz — COMMENTED** [review]
+1

**Pull request merged**